SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -7320,231 +7308,195 @@ void CEulerSolver::SetActDisk_BCThrust(CGeometry *geometry, CSolver **solver_con
-      Vel_Infty[1] = sin(AoA)*Vel_Infty_Mag;
-    }
-    if (nDim == 3) {
-      Vel_Infty[0] = cos(AoA)*cos(Beta)*Vel_Infty_Mag;
-      Vel_Infty[1] = sin(Beta)*Vel_Infty_Mag;
-      Vel_Infty[2] = sin(AoA)*cos(Beta)*Vel_Infty_Mag;
-    }
-    
-    /*--- Store the new freestream velocity vector for the next iteration ---*/
-    
-    for (iDim = 0; iDim < nDim; iDim++) {
-      Velocity_Inf[iDim] = Vel_Infty[iDim];
-    }
-    
-    /*--- Only the fine mesh stores the updated values for velocity in config ---*/
-    
-    if (iMesh == MESH_0) {
-      for (iDim = 0; iDim < nDim; iDim++)
-        config->SetVelocity_FreeStreamND(Vel_Infty[iDim], iDim);
+  AoA = config->GetAoA();
+
+  /* --- If AoA needs to be updated, calculate change in AoA --- */
+
+  if (Update_AoA && Output) {
+
+    /*--- Estimate the increment in AoA based on dCL_dAlpha (degrees) ---*/
+    if (!CL_Converged){
+      AoA_inc = (1.0/dCL_dAlpha)*(Target_CL - Total_CL);
     }
-    
+
     /*--- Output some information to the console with the headers ---*/
     
-    if ((rank == MASTER_NODE) && (iMesh == MESH_0) && write_heads && !config->GetDiscrete_Adjoint()) {
-      Old_AoA = config->GetAoA() - AoA_inc*(180.0/PI_NUMBER);
-      
+    if ((rank == MASTER_NODE) && (iMesh == MESH_0) && !config->GetDiscrete_Adjoint()) {
       cout.precision(7);
       cout.setf(ios::fixed, ios::floatfield);
       cout << endl << "----------------------------- Fixed CL Mode -----------------------------" << endl;
       cout << "CL: " << Total_CL;
       cout << " (target: " << config->GetTarget_CL() <<")." << endl;
-      cout.precision(4);
-      cout << "Previous AoA: " << Old_AoA << " deg";
-      cout << ", new AoA: " << config->GetAoA() << " deg." << endl;
+      if (CL_Converged){
+        cout << "CL converged to within "<< config->GetCauchy_Eps() <<", not changing AoA" << endl;
+      }
+      else {
+        cout.precision(4);
+        cout << "Previous AoA: " << AoA << " deg";
+        cout << ", new AoA: " << AoA + AoA_inc << " deg." << endl;
+      }
       cout << "-------------------------------------------------------------------------" << endl << endl;
     }
 
   }
 
-	unsigned long Iter_dCL_dAlpha = config->GetIter_dCL_dAlpha();
-
-  if ((config->GetnInner_Iter()-Iter_dCL_dAlpha == InnerIter) && Output) {
+  /* --- Start Finite-differencing routine --- */
 
-    AoA_old = config->GetAoA();
+  if (Start_AoA_FD && Output && ((InnerIter - 1) == Iter_Update_AoA)) {
 
-    if (config->GetnInner_Iter()-Iter_dCL_dAlpha == InnerIter) {
-      Wrt_Con_Freq = SU2_TYPE::Int(su2double(config->GetIter_dCL_dAlpha())/10.0);
-      config->SetWrt_Con_Freq(Wrt_Con_Freq);
-      Total_CD_Prev = Total_CD;
-      Total_CL_Prev = Total_CL;
-      Total_CMx_Prev = Total_CMx;
-      Total_CMy_Prev = Total_CMy;
-      Total_CMz_Prev = Total_CMz;
-      AoA_inc = 0.001;
-      AoA_FD_Change = true;
+    Wrt_Con_Freq = SU2_TYPE::Int(su2double(Iter_dCL_dAlpha)/10.0);
```

Comment:
Can you please check if all these casts are required? It seems Iter_dCL_dAlpha is of integer type, so you can probably just divide by 10.

Reply:
Good call, not needed

Reply:
There were a few other places.